soccer/gameplay/tactics/defensive_forward.py

```diff
@@ -0,0 +1,216 @@
+            # TODO: Shift to mark point when "get_block_pos" is fixed
+
+        self.defenders[2] = skills.mark.Mark()
+        self.add_subbehavior(self.defenders[2], self.names[2], required=True)
+        self.defenders[2].mark_point = self.zone_def_pos
+
+        # Reset time since call
+        self.time_since_call = 0
+
+    # Continue updating the mark positions
+    def execute_blocking(self):
+        self.time_since_call = (self.time_since_call + 1) % self.cache_amnt
+
+        # You don't want to test if its 0 because it would execute instantly
+        if (self.time_since_call % self.cache_amnt == self.cache_amnt - 1):
+            self.zone_def_pos, self.mark_bots[0], self.mark_bots[1] = \
+                evaluation.defensive_positioning.find_defense_positions(
+                    [self.defenders[0].robot, self.defenders[1].robot])
+
+            for i in range(0, 2):
+                if (self.defenders[i].mark_robot not in self.mark_bots):
+                    self.defenders[i].mark_robot = self.mark_bots[i]
+                # TODO: Shift to mark point when "get_block_pos" is fixed
+
+            self.defenders[2].mark_point = self.zone_def_pos
+
+    def on_exit_blocking(self):
+        self.remove_all_subbehaviors()
+
+    # Create collector method and then
+    def on_enter_collecting(self):
+        self.zone_def_pos, self.mark_bots[0], self.mark_bots[1] = \
+            evaluation.defensive_positioning.find_defense_positions()
+
+        # Take closest robot to collect
+        # Leave other two to move into primary blocking positions
+        self.collector = skills.capture.Capture()
+        self.add_subbehavior(self.collector, 'collector', required=True)
+
+        for i in range(0, 2):
+            self.defenders[i] = skills.mark.Mark()
+            self.add_subbehavior(self.defenders[i],
+                                 self.names[i],
+                                 required=True)
+            self.defenders[i].mark_robot = self.mark_bots[i]
+
+    def on_exit_collecting(self):
+        self.remove_all_subbehaviors()
+        self.collector = None
+
+    # Uses their predicted kick direction to block
+    def get_block_pos(self, bot):
+        # Get predicted angle of shot
+        # Get goal to bot
+        # Place point x dist away
+        predicted = evaluation.defensive_positioning.predict_kick_direction(
+            bot)
+        actual = bot.angle
+
+        angle = self.block_angle_coeff * predicted + (
+            1 - self.block_angle_coeff) * actual
+
+        x = math.cos(angle)
+        y = math.sin(angle)
+        pos = robocup.Point(x, y).normalized() + bot.pos
+
+        return None  # pos * self.block_dist
+
+    def we_have_ball(self):
+        has_ball = False
+        for bot in main.our_robots():
+            has_ball = has_ball or robot_has_ball(bot)
+        return has_ball
+
+    # Wether any robot can collect the ball before the opponent
+    def within_range(self):
```

Comment:
Are you saying the function is too restricted and doesn't return true often enough?